scripts/webidl/convert.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+const fs = require("fs");
+const path = require("path");
+
+const Q = require("q");
+const recursive = require("recursive-readdir");
+const webidl2js = require("webidl2js");
+
+const UTIL_PATH = "lib/jsdom/living/generator/util.js";
+
+function readConcatenatedSource(files) {
+  return Q.all(files.map(function (f) {
+    return Q.nfcall(fs.readFile, f, { encoding: "utf8" });
+  })).then(function (sources) {
+    let src = "";
+    for (let i = 0; i < sources.length; ++i) {
+      src += sources[i];
+    }
+    return src;
+  });
+}
+
+function generateClasses(src, outputDir, implDir, utilPath) {
+  webidl2js.generate(src, outputDir, implDir, { suppressErrors: true, implSuffix: "-impl", utilPath: utilPath });
+}
+
+function doConversion(inputPath) {
+  let isDir;
+
+  Q.nfcall(fs.stat, inputPath)
```

Comment:
return this promise, then the last line becomes `doConversion(...).done()`, to ensure errors get propagated to the console.